packages/react/src/components/GroupedList/GroupedList.test.tsx

```diff
@@ -307,4 +307,31 @@ describe('GroupedList', () => {
     expect(onRenderCheckboxMock).toHaveBeenCalledTimes(1);
     expect(onRenderCheckboxMock.mock.calls[0][0]).toEqual({ checked: false, theme: getTheme() });
   });
+
+  it('re-renders when items change back to the initial items', () => {
+    const initialItems: Array<{ key: string }> = [{ key: 'initial' }];
+    const nextItems: Array<{ key: string }> = [{ key: 'changed' }];
+    const _groups: Array<IGroup> = [
+      {
+        count: 1,
+        hasMoreData: true,
+        isCollapsed: false,
+        key: 'group0',
+        name: 'group 0',
+        startIndex: 0,
+        level: 0,
+      },
+    ];
+
+    function _onRenderCell(nestingDepth: number, item: { key: string }, itemIndex: number): JSX.Element {
+      const id = `rendered-item-${item.key}`;
+      return <div id={id} />;
+    }
+
+    const wrapper = mount(<GroupedList items={initialItems} groups={_groups} onRenderCell={_onRenderCell} />);
+    wrapper.setProps({ items: nextItems });
+    wrapper.setProps({ items: initialItems });
+
+    expect(wrapper.contains(<div id="rendered-item-initial" />)).toEqual(true);
```

Comment:
I like this test but I'm not sure I trust that it's verifying its assertion. The initial render of items is `initialItems` and the assertion at the end is that `GroupedList` rendered `initialItems` but since we never assert against the intermediate renders we don't actually know if we're currently asserting against the first render or the last prop update.

I suggest the following changes:

```tsx
const wrapper = mount(<GroupedList items={initialItems} groups={_groups} onRenderCell={_onRenderCell} />);
// Assert against the initial render
expect(wrapper.contains(<div id="rendered-item-initial" />)).toEqual(true);

// Assert against the first update
wrapper.setProps({ items: nextItems });
expect(wrapper.contains(<div id="rendered-item-changed" />)).toEqual(true);

// Assert that initialItems is re-rendered
wrapper.setProps({ items: initialItems });
expect(wrapper.contains(<div id="rendered-item-initial" />)).toEqual(true);
```

Reply:
> I like this test but I'm not sure I trust that it's verifying its assertion

Agreed, there is an implicit assertion that it does re-render with the first props change.